src/retrievals.ts

```diff
@@ -47,36 +50,94 @@ export class Class {
         this.script = specs?.script
         if (specs?.url) {
             this.url = specs.url
-            let parts = parseURL(specs.url)
-            this.schema = parts[0]
-            if (parts[1] !== "") this.authority = parts[1]
-            if (parts[2] !== "") this.path = parts[2]
-            if (parts[3] !== "") this.query = parts[3]
+            if (!utils.isWildcard(specs.url)) {
+                let parts = utils.parseURL(specs.url)
+                this.schema = parts[0]
+                if (parts[1] !== "") this.authority = parts[1]
+                if (parts[2] !== "") this.path = parts[2]
+                if (parts[3] !== "") this.query = parts[3]
+            }
         }
         this.argsCount = Math.max(
-            getMaxArgsIndexFromString(specs?.url),
-            getMaxArgsIndexFromString(specs?.body),
-            ...this.headers.map(header => getMaxArgsIndexFromString(header[1])),
+            utils.getMaxArgsIndexFromString(specs?.url),
+            utils.getMaxArgsIndexFromString(specs?.body),
+            ...this.headers.map(header => utils.getMaxArgsIndexFromString(header[1])),
             specs?.script?._countArgs() || 0,
         )
         this.tuples = specs?.tuples
     }
+    /**
+     * Creates clones of this retrieval where all occurences of the specified parameter 
+     * are replaced by the given values. Fails if the retrieval refers no parameters,
+     * or if the specified index is not referred by it. 
+     * @param argIndex Index of the parameter upon which the new instances will be created.
+     * @param values Values used for the creation of the new retrievals. 
+     * @returns An array with as many retrievals as spawning values were specified.
+     */
+    public spawn(argIndex: number, values: string[]): Class[] {
+        let spawned: Class[] = []
+        if (this.argsCount == 0) {
+            throw new EvalError(`\x1b[1;33mRetrieval: cannot spawn over unparameterized retrieval\x1b[0m`);
+        } else if (argIndex > this.argsCount) {
+            throw new EvalError(`\x1b[1;33mRetrieval: spawning parameter index out of range: ${argIndex} > ${this.argsCount}\x1b[0m`);
+        }
+        values.map(value => {
+            let headers: any 
+            if (this.headers) {
+                this.headers?.map(header => {
+                    headers[
+                        utils.spliceWildcards(header[0], argIndex, value, this.argsCount)
+                    ] = utils.spliceWildcards(header[1], argIndex, value, this.argsCount)
+                })
+            }
+            const script: Script | undefined = this.script?._spliceWildcards(argIndex, value);
+            spawned.push(new Class(this.method, {
+                url: utils.spliceWildcards(this.url, argIndex, value, this.argsCount),
+                body: utils.spliceWildcards(this.body, argIndex, value, this.argsCount),
+                headers, script,
+            }))
+        })
+        return spawned
+    }
```

Comment:
As we are already using a `map` instead of a `for` statement, we can use reduce to avoid declaring extra variables.

```suggestion
        const spawned: Class[] = values.reduce((acc, value) => { 
             this.headers = this.headers?.reduce((headers, header) => {
                headers[
                    utils.spliceWildcards(header[0], argIndex, value, this.argsCount)
                ] = utils.spliceWildcards(header[1], argIndex, value, this.argsCount)
                return headers
            }, this.headers)
            
            const script: Script | undefined = this.script?._spliceWildcards(argIndex, value);
            acc.push(new Class(this.method, {
                url: utils.spliceWildcards(this.url, argIndex, value, this.argsCount),
                body: utils.spliceWildcards(this.body, argIndex, value, this.argsCount),
                headers, script,
            }))
        })
     
    }
```

Reply:
Not valid, as `this.headers` must not be altered.